cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -128,6 +129,7 @@ func Run(s *options.CMServer) error {
 	}
 
 	kubeconfig.ContentConfig.ContentType = s.ContentType
+	kubeconfig.ContentConfig.GroupVersion = &unversioned.GroupVersion{Group: batch.GroupName, Version: "v2alpha1"}
```

Comment:
On second thought, can you move this piece and initiate the kubeclient with this version only when `batch/v2alpha1` is enabled?